### PR TITLE
Clesaec/tdi 44213 workday raas login

### PR DESCRIPTION
**Pull request by clesaec**
link to https://jira.talendforge.org/browse/TDI-44213

Add possibility to login with login/pwd (in addition to clientId/clientSecret) with restriction to RAAS.

### CODE REVIEW

File: workday/src/main/java/org/talend/components/workday/datastore/WorkdayDataStore.java [hunk 1]
```diff
@@ -12,63 +12,62 @@
...
-@Documentation(WorkdayDataStore.NAME)
+@Documentation("DataStore for workday connector")
 public class WorkdayDataStore implements Serializable {
 
     private static final long serialVersionUID = -8628647674176772061L;
 
-    public static final String NAME = "WorkdayDataStore";
+    public enum AuthenticationType {
+        ClientId,
+        Login;
```
Comment (rlecomte-talend) [inline workday/src/main/java/org/talend/components/workday/datastore/WorkdayDataStore.java:47]: enum values should be uppercase

Comment (clesaec) [reply workday/src/main/java/org/talend/components/workday/datastore/WorkdayDataStore.java]: changed

File: workday/src/main/resources/org/talend/components/workday/datastore/Messages.properties [hunk 2]
```diff
@@ -13,3 +13,15 @@ WorkdayDataStore.endpoint._placeholder=https://api.workday.com
...
+AuthenticationType.LOGIN._displayName   =Login Authentication Mode
+
+Workday.datastore.WorkdayDataStore._displayName=Workday Connection
+
+WorkdayDataStore.loginForm._displayName=Login Authentication Form
+
+UserFormForReport.endpointPattern._displayName=Service Endpoint
+UserFormForReport.login._displayName=User Login
+UserFormForReport.password._displayName=User Password
+UserFormForReport.tenantAlias._displayName=Tenant Alias
```
Comment (rlecomte-talend) [inline workday/src/main/resources/org/talend/components/workday/datastore/Messages.properties:27]: For all labels : only the first word should be capitalized

File: workday/src/main/resources/org/talend/components/workday/Messages.properties [hunk 3]
```diff
@@ -5,3 +5,5 @@ Workday.dataset.WorkdayDataSet._displayName=Workday Data Set
 Workday.actions.healthcheck.WORKDAY_HEALTH_CHECK._displayName=Check connection
 Workday.actions.validation.WORKDAY_VALIDATION_URL_PROPERTY._displayName=URL Kind Validation
 
+Workday.actions.suggestions.WORKDAY_MODE_SUPPORTED._displayName=Supported Authentication Modes
```
Comment (ypiel-talend) [inline workday/src/main/resources/org/talend/components/workday/Messages.properties:8]: are those two labels used ?

File: workday/src/test/java/org/talend/components/workday/input/RAASLoginProducerTest.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import org.talend.components.workday.datastore.WorkdayDataStore;
+import org.talend.components.workday.datastore.WorkdayDataStore.AuthenticationType;
+import org.talend.components.workday.service.WorkdayReaderService;
+import org.talend.sdk.component.api.service.Service;
+import org.talend.sdk.component.junit.http.junit5.HttpApi;
+import org.talend.sdk.component.junit5.WithComponents;
+
+@HttpApi(useSsl = true)
+@WithComponents("org.talend.components.workday")
+class RAASLoginProducerTest /* extends WorkdayBaseTest */ {
```
Comment (ypiel-talend) [inline workday/src/test/java/org/talend/components/workday/input/RAASLoginProducerTest.java:31]: Remove commented extends ?

File: workday/src/main/java/org/talend/components/workday/service/UIActionService.java [hunk 5]
```diff
@@ -44,10 +45,10 @@
     @HealthCheck(HEALTH_CHECK)
     public HealthCheckStatus validateConnection(@Option final WorkdayDataStore dataStore) {
         try {
-            service.getAccessToken(dataStore);
-            return new HealthCheckStatus(OK, i18n.healthCheckOk());
+            this.service.getAccessToken(dataStore);
+            return new HealthCheckStatus(OK, this.i18n.healthCheckOk());
         } catch (Exception e) {
-            return new HealthCheckStatus(KO, i18n.healthCheckFailed("msg", e.getMessage()));
+            return new HealthCheckStatus(KO, this.i18n.healthCheckFailed("msg", e.getMessage()));
```
Comment (ypiel-talend) [inline workday/src/main/java/org/talend/components/workday/service/UIActionService.java:51]: 2 parameters for healthCheckFailed but only one placeholder in the resource : https://github.com/Talend/connectors-se/blob/8c37b58b77277e6fba5a2418d348223a5596fb3c/workday/src/main/resources/org/talend/components/workday/service/Messages.properties#L3

File: workday/src/main/resources/org/talend/components/workday/dataset/Messages.properties [hunk 6]
```diff
@@ -1,5 +1,6 @@
 WorkdayDataset._displayName=Workday Data Set
 WorkdayDataSet.mode._displayName=Mode
+WorkdayDataSet.mode2._displayName=Mode2
```
[the remark this answers is not in the document]
Comment (clesaec) [reply workday/src/main/resources/org/talend/components/workday/dataset/Messages.properties]: yes, removed

File: workday/src/main/resources/org/talend/components/workday/datastore/Messages.properties (hunk 2 above)
Comment (clesaec) [reply workday/src/main/resources/org/talend/components/workday/datastore/Messages.properties]: updated

File: workday/src/test/java/org/talend/components/workday/input/RAASLoginProducerTest.java (hunk 4 above)
Comment (clesaec) [reply workday/src/test/java/org/talend/components/workday/input/RAASLoginProducerTest.java]: done

File: workday/src/main/java/org/talend/components/workday/service/UIActionService.java (hunk 5 above)
Comment (clesaec) [reply workday/src/main/java/org/talend/components/workday/service/UIActionService.java]: first parameter removed

File: workday/src/main/resources/org/talend/components/workday/Messages.properties (hunk 3 above)
Comment (clesaec) [reply workday/src/main/resources/org/talend/components/workday/Messages.properties]: WORKDAY_MODE_SUPPORTED removed

File: workday/src/main/java/org/talend/components/workday/datastore/WorkdayDataStore.java [hunk 7]
```diff
@@ -12,34 +12,49 @@
...
-@GridLayout(names = GridLayout.FormType.ADVANCED, value = { @GridLayout.Row("authEndpoint"), @GridLayout.Row("endpoint") })
+@GridLayout(names = GridLayout.FormType.ADVANCED, //
+        value = { @GridLayout.Row({ "authEndpoint", "loginForm" }), @GridLayout.Row("endpoint") })
 @Checkable(UIActionService.HEALTH_CHECK)
-@Documentation(WorkdayDataStore.NAME)
+@Documentation("DataStore for workday connector")
 public class WorkdayDataStore implements Serializable {
 
-    private static final long serialVersionUID = -8628647674176772061L;
+    private static final long serialVersionUID = -8628647674176772061L; // 8254399046469388027L;
```
Comment (ypiel-talend) [inline workday/src/main/java/org/talend/components/workday/datastore/WorkdayDataStore.java:46]: Remove comment

Comment (clesaec) [reply workday/src/main/java/org/talend/components/workday/datastore/WorkdayDataStore.java]: done

**Pull request closed without merging**